src/lib.ts

```diff
@@ -0,0 +1,180 @@
+import 'zx/globals';
+import xdgAppPaths from "xdg-app-paths";
+import path from "node:path";
+import fs from 'fs/promises'
+import { Auth, Config } from './types.js';
+import { deepmerge } from "deepmerge-ts";
+import login from './login.js';
+import packageJson from '../package.json' assert { type: "json" };
+
+let currentTries = 0;
+
+export const VERSION = packageJson.version;
+export const MAX_TRIES = 3;
+export const BUILD_DIR = 'app';
+export const CONFIG_PATH = xdgAppPaths("webstudio").config();
+export const CONFIG_FILE = path.join(CONFIG_PATH, "config.json");
+export let pm: string | null = null;
+export const supportedBuildTypes = ['remix-app-server', 'express', 'architect', 'flyio', 'netlify', 'vercel', 'cloudflare-pages', 'cloudflare-workers', 'deno'];
+
+const HELP = `Usage:
+    $ webstudio commands [flags...]
+  Commands:
+    login <shared link>             Login to Webstudio with shared link
+    sync <projectId>                Download a project's site data
+    build [projectId]               Build a site
+  Flags:
+    --type, -t                      Build type chosen during build command (default: remix-app-server) 
+                                    (options: ${supportedBuildTypes.join(', ')})
+    --debug                         Enable debug mode
+    --help, -h                      Show this help message
+    --version, -v                   Show the version of this script
+`;
+export const showHelp = () => {
+    console.log(HELP);
+}
+export const detectPackageManager = async () => {
+    const originalverbose = $.verbose;
+    $.verbose = false;
+    try {
+        const pnpm = await $`pnpm --version`.exitCode;
+        const yarn = await $`yarn --version`.exitCode;
+        const npm = await $`npm --version`.exitCode;
```

Comment:
you are executing all of these even though you could potentially return after the first match